sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java

```diff
@@ -191,12 +191,16 @@ private <DTO> boolean pseudonymizeDto(
 			return false;
 		}
 
-		boolean didPseudonymization = false;
+		boolean didPersonalOrSensitiveDataPseudonymization = false;
 
 		for (Field field : pseudonymizableFields) {
-			if (!getFieldAccessCheckers(inJurisdiction).isAccessible(field, pseudonymizeMandatoryFields)) {
+			FieldAccessCheckers fieldAccessCheckers = getFieldAccessCheckers(inJurisdiction);
+			if (!fieldAccessCheckers.isAccessible(field, pseudonymizeMandatoryFields)) {
 				pseudonymizeField(dto, field, defaultPseudonymizerClass);
-				didPseudonymization = true;
+				// only personal and sensitive data pseudonymization needs special handling on the client side
+				// other not accessible data is hidden on the client side, so just cleanup and don't mark the DTO as pseudonymized
+				didPersonalOrSensitiveDataPseudonymization = !fieldAccessCheckers
```

Comment:
I think this has to be "|=" instead of "="

Reply:
Better: Only do logic when didPersonalOrSensitiveDataPseudonymization is still false